crates/turbopack-ecmascript/src/chunk/mod.rs

```diff
@@ -739,6 +739,21 @@ impl GenerateSourceMap for EcmascriptChunkContent {
     fn generate_source_map(self_vc: EcmascriptChunkContentVc) -> SourceMapVc {
         self_vc.code().generate_source_map()
     }
+
+    #[turbo_tasks::function]
+    async fn by_section(&self, section: StringVc) -> Result<OptionSourceMapVc> {
+        let section = section.await?;
+        for entry in self.module_factories.iter() {
```

Comment:
Maybe store as IndexMap by is for faster lookup

Reply:
This is a nested data structure unfortunately, and it's a bit difficult to update the several structs that implement that.